department-of-veterans-affairs/caseflow
# Pamatya/appeals 34662

Resolves [APPEALS-34662](https://jira.devops.va.gov/browse/APPEALS-34662)

# Description
Please explain the changes you made here.
Added Validation error message in RHFCheckbox component so that error message can be handled.
Yup validation was added to verify if Specific Events / Actions radio is selected atleast one of the Checkbox is selected. 
Added a jest test to verify test. 

## Acceptance Criteria
- [ ] Validation error message for Specific Events / Actions has to be selected. 
- [ ] Should not break other validations or other functionality. i.e if Specific Events/ Actions radio is not selected then we should see the console data log as expected. 

## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-12345) or list them below
2. Go To Generate Task Report page
3. Select Event / Action in Report Type dropdown. 
4. Select Radio -> Specific Events / Actions 
5. Click on Generate task report button on the button, Check if the validation is present. 
6.  select random Checkbox under Specific event / Actions and click on Generate task report, validation message should go away. 
7. Select -> All Events / Actions and click on Generate task report button and  validation message should not be present. 


# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/122557351/6ce45313-ad3b-4a0e-b506-be4f9e4be151)

## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
